### PR TITLE
Added Audience Parameter and Condition for proxy setup

### PR DESCRIPTION
**Pull request by vaibhavk**
[WIP]: Fixes Bugzilla [Bug 1517015](https://bugzilla.mozilla.org/show_bug.cgi?id=1517015)
@djmitche please review

### CODE REVIEW

**Comment by djmitche** [comment]
@vaibhavk it looks like the tests are failing..

**Comment by vaibhavk** [comment]
:100: tests passing now

File: cmd/websocktunnel/main.go [hunk 1]
```diff
@@ -28,6 +28,7 @@ Environment:
  TASKCLUSTER_PROXY_SECRET_A                  JWT secret
  TASKCLUSTER_PROXY_SECRET_B                  alternate JWT secret
  SYSLOG_ADDR                                 address to which to send syslog output
+ AUDIENCE(optional;default to "wst-servers") JWT 'audience' claim
```
Comment (djmitche) [inline cmd/websocktunnel/main.go]: I'd rather not have a default. It's better to be explicit than spend time debugging only to finally see this note and say "oh, *that's* the value I should be using for audience!"

Comment (vaibhavk) [reply cmd/websocktunnel/main.go]: Sure, will remove the defaults

**Comment by vaibhavk** [comment]
Yes, I'll be adding tests

**Comment by vaibhavk** [comment]
@djmitche @petemoore Looks like linting was failing because of latest pull of golang image.

File: .taskcluster.yml [hunk 2]
```diff
@@ -94,7 +94,7 @@ tasks:
       workerType: github-worker
       payload:
         maxRunTime: 3600
-        image: golang
```
Comment (djmitche) [inline .taskcluster.yml]: :+1: this matches the README's declaration of support

File: cmd/websocktunnel/main.go [hunk 3]
```diff
@@ -28,6 +28,7 @@ Environment:
  TASKCLUSTER_PROXY_SECRET_A                  JWT secret
  TASKCLUSTER_PROXY_SECRET_B                  alternate JWT secret
  SYSLOG_ADDR                                 address to which to send syslog output
+ AUDIENCE									 JWT 'audience' claim
```
Comment (djmitche) [inline cmd/websocktunnel/main.go]: did the whitespace get mixed up here?

Comment (vaibhavk) [reply cmd/websocktunnel/main.go]: I should fix this quick

Comment (vaibhavk) [reply cmd/websocktunnel/main.go]: Looks like it's the github's tab indentation that's causing it

**Comment by djmitche** [comment]
Thanks for the quick fixes!  @petemoore should be back tomorrow and can hopefully have a look.

File: wsproxy/proxy_test.go [hunk 4]
```diff
@@ -1195,3 +1195,62 @@ func TestProxyWebSocketPath(t *testing.T) {
...
+	_ = conn.Close()
+
+	wstoken := token("wsworker", "INVALID-AUD", []byte("test-secret"))
+	header.Set("Authorization", "Bearer "+wstoken)
+	header.Set("x-websocktunnel-id", "wsworker")
+
+	if _, _, err := websocket.DefaultDialer.Dial(wsURL, header); err == nil {
+		t.Fatal("Wrong Audience Value accepted")
+	}
+}
```
Comment (petemoore) [inline wsproxy/proxy_test.go]: Should we add a test that using the correct audience does not generate an error?

Comment (vaibhavk) [reply wsproxy/proxy_test.go]: Line 1239-1247 contains the test with correct audience value

**Pull request merged**